### ISSUE TITLE
Social follows feed: Buttons not highlighted properly when Approved. Ignore not working at all

### ISSUE THREAD

**Issue by petervojtek** [PowerlineApp/powerline-mobile#234]

**Comment by petervojtek** [comment]
ok, Approve is properly highlighted.

For Ignore button I am blocked by broken API doc.

**Comment by petervojtek** [comment]
@jterps08 , what is the Ignore button supposed to do?

there does not seem to be any API endpoint to ignore follow request.

<img width="994" alt="screen shot 2016-09-15 at 14 42 31" src="https://cloud.githubusercontent.com/assets/225506/18550175/c89e30c6-7b52-11e6-839c-ed00ee4a9396.png">

**Comment by jterps08** [comment]
The original intended behavior was this:
User A receives follow request from User B
User A sees two buttons Approve and Ignore.
User taps Approve first, and button changes to Unapprove (and marked as read) Ignore changes to Follow. 
User taps Follow and button changes to Unfollow.
User A now sees two buttons Unapprove and Unfollow.

If User A taps Ignore first, the notification is marked as read and the Ignore button changes to Follow.

**Comment by jterps08** [comment]
API doc should be available. Please let us know if there's anything else blocking this issue that Igor needs to work on.

**Comment by petervojtek** [comment]
@jterps08 , the requested button behaviour is very complicated and prone to errors. for example it is not clear which buttons whould be visible when I click Approve, and then Unapprove.

Also there is problem that when you Approve user, then Unapprove, and then try to Approve again, the second Approve fails will 404 (Not Found) because the first approve changed some state on server and the "Approval" is no longer available there.

the rules to display or not display some button are too complicated because the require up to four conditions to be checked on frontend:
- is ignored
- is approved
- is followed
- is read

this is hard and time consuming to test, because there are many scenarios when there are up to 5 buttons (follow, unfollow, approve, unapprove, ignore) and 4 conditions.

Also we have currently problem with proper detection that an social notification has been read. 
Currently the business logic to detect if notification is read/unread is implemented on front-end only and build on [local caching](https://github.com/PowerlineApp/powerline-mobile/blob/develop/www/js/services/social-activity/social-activity-tab-manager.js#L8) and parsing timestamp from [server-headers](https://github.com/PowerlineApp/powerline-mobile/blob/develop/www/js/services/social-activity.js#L46), which is really terrible approach, and despite the effort it is not actually working due to bug #293.

Also, I searched the API but I have not discovered which API endpoint should I call when Ignore button is clicked -- UPDATE -- ok, now I understand that it just means "the notification is marked as read and the Ignore button changes to Follow."

To sum up, to fix it quickly I suggest following solution which will not require any changes on backend and requires only to remove Ignore button from the current implementation:
1. When follow request comes, there are two buttons: Approve and Follow. 
2. Which you click Approve, it will change to Unapprove, and vice versa.
3. When you click Follow, it will change to Unfollow, and vice versa.

and later we should move all the business logic to backend and frontend will be told via attribute like `buttons: ['Approve', 'Ignore']` attached on social request which buttons should be actually displayed.

**Comment by jterps08** [comment]
works for me

**Comment by petervojtek** [comment]
@jterps08 , it sort of works.. I mean the Approve/Unapprove/Follow and Unfollow buttons really work and currently I discovered that also Ignore button sends request to this endpoint [`PUT https://api-dev.powerli.ne/api-doc#put--api-v2-social-activities-{id}`](https://api-dev.powerli.ne/api-doc#put--api-v2-social-activities-{id}). 

but the rules on how the buttons appear and disappear are not that sophisticated as you described above. I am not sure if you want me to pay attention to it (it will be time consuming to test it), or the current behaviour is safisfactory to ship it (and adjust it properly later).

To sum up, the important use cases already work -- i.e. Approve, Unapprove, Follow and Unfollow.

**Comment by jterps08** [comment]
Let's close this. We can come back as an enhancement.
